### PR TITLE
virsh_event.py: Add crash panic event

### PR DESCRIPTION
**Pull request by yanqzhan**
Add 'crash panic' lifecycle event.

Signed-off-by: Yanqiu Zhang <yanqzhan@redhat.com>

### CODE REVIEW

**Comment by yanqzhan** [comment]
Local run passed:
# avocado run --vt-type libvirt virsh.event..crash_panic
JOB ID     : 7e1d8a091c1b845bef64d2bbe6c6c2f3db6776b5
JOB LOG    : /root/avocado/job-results/job-2019-03-14T22.47-7e1d8a0/job.log
 (1/1) type_specific.io-github-autotest-libvirt.virsh.event.positive_test.virsh_event.test_events.lifecycle_events.crash_panic.no_timeout.loop: PASS (91.83 s)
RESULTS    : PASS 1 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB TIME   : 93.98 s

And all other virsh.event cases passed: http://pastebin.test.redhat.com/738069 (1month available)

File: libvirt/tests/src/virsh_cmd/domain/virsh_event.py [hunk 1]
```diff
@@ -197,6 +201,32 @@ def trigger_events(dom, events_list=[]):
                                                     " Shutdown Finished")
                     expected_events_list.append("'lifecycle' for %s:"
                                                 " Stopped Shutdown")
+                elif event == "crash":
+                    if not vmxml.xmltreefile.find('devices').findall('panic'):
+                        # Set panic device
+                        panic_dev = Panic()
+                        panic_dev.model = 'isa'
```
Comment (chunfuwen) [inline libvirt/tests/src/virsh_cmd/domain/virsh_event.py]: suggest put those configuration into cfg file

File: libvirt/tests/src/virsh_cmd/domain/virsh_event.py [hunk 2]
```diff
@@ -197,6 +201,32 @@ def trigger_events(dom, events_list=[]):
...
+                        panic_dev.addr_type = 'isa'
+                        panic_dev.addr_iobase = '0x505'
+                        vmxml.add_device(panic_dev)
+                    vmxml.on_crash = "coredump-restart"
+                    vmxml.sync()
+                    dom.start()
+                    session = dom.wait_for_login()
+                    # Stop kdump in the guest
+                    session.cmd("service kdump stop", ignore_all_errors=True)
```
Comment (chunfuwen) [inline libvirt/tests/src/virsh_cmd/domain/virsh_event.py]: maybe use systemctl for future migration

File: libvirt/tests/src/virsh_cmd/domain/virsh_event.py [hunk 3]
```diff
@@ -197,6 +201,32 @@ def trigger_events(dom, events_list=[]):
...
+                    # Enable sysRq
+                    session.cmd("echo 1 > /proc/sys/kernel/sysrq")
+                    try:
+                        # Crash the guest
+                        session.cmd("echo c > /proc/sysrq-trigger", timeout=60)
+                    except (ShellTimeoutError, ShellProcessTerminatedError):
+                        pass
```
Comment (chunfuwen) [inline libvirt/tests/src/virsh_cmd/domain/virsh_event.py]: logging.info(xx)

File: libvirt/tests/src/virsh_cmd/domain/virsh_event.py (hunk 1 above)
Comment (yanqzhan) [reply libvirt/tests/src/virsh_cmd/domain/virsh_event.py]: Done.

File: libvirt/tests/src/virsh_cmd/domain/virsh_event.py (hunk 2 above)
Comment (yanqzhan) [reply libvirt/tests/src/virsh_cmd/domain/virsh_event.py]: Done.

File: libvirt/tests/src/virsh_cmd/domain/virsh_event.py (hunk 3 above)
Comment (yanqzhan) [reply libvirt/tests/src/virsh_cmd/domain/virsh_event.py]: I just want to ignore these two kinds of error, no info need to print.

**Comment by yanqzhan** [comment]
4 more lines are modified:
(1)For each domain, each check index will be started from begining, but not the tail after last domain.
+            if event in expected_events_list[0]:
+                event_idx = 0
(2)Should print expected event to compare with 'Actual events'.
+            logging.info("Expected event: %s", event_str)
(3)No need to print output again, it's same with 'Actual events', and even wants, should be  output[event_idx] but not  output[event_idx:], which involved much redundant duplicated logs.
-                logging.debug("Output event: %s", output[event_idx:])

File: libvirt/tests/src/virsh_cmd/domain/virsh_event.py (hunk 3 above)
Comment (dzhengfy) [reply libvirt/tests/src/virsh_cmd/domain/virsh_event.py]: Agree with chunfuwen. We'd better print any exception message which will benefit our debugging.

**Comment by dzhengfy** [comment]
@yanqzhan I am fine with your changes in comment 13. Thanks.

**Comment by yanqzhan** [comment]
1 more update: Add rm files under dump_file directory to avoid occupying too much host storage.
**+**     dump_path = '/var/lib/libvirt/qemu/dump'
...
**+** if os.path.exists(dump_path):
**+** shutil.rmtree(dump_path)

File: libvirt/tests/src/virsh_cmd/domain/virsh_event.py (hunk 3 above)
Comment (yanqzhan) [reply libvirt/tests/src/virsh_cmd/domain/virsh_event.py]: Done.

**Comment by yanqzhan** [comment]
@chunfuwen Pls help review again.

**Pull request merged**